core/domain/suggestion_services.py

```diff
@@ -0,0 +1,382 @@
+# limitations under the License.
+
+"""Funtions to create, accept, reject, update and perform other operations on
+suggestions.
+"""
+
+from core.domain import feedback_services
+from core.domain import suggestion_registry
+from core.domain import user_services
+from core.platform import models
+import feconf
+import utils
+
+(feedback_models, suggestion_models) = models.Registry.import_models([
+    models.NAMES.feedback, models.NAMES.suggestion])
+
+DEFAULT_SUGGESTION_THREAD_SUBJECT = 'Suggestion from a user'
+DEFAULT_SUGGESTION_THREAD_INITIAL_MESSAGE = ''
+
+
+def create_suggestion(
+        suggestion_type, target_type, target_id, target_version_at_submission,
+        author_id, change_cmd, score_category, description,
+        assigned_reviewer_id, final_reviewer_id):
+    """Creates a new SuggestionModel and the corresponding FeedbackThread.
+
+    Args:
+        suggestion_type: str. The type of the suggestion.
+        target_type: str. The target entity being edited.
+
+        (The above 2 parameters should be one of the constants defined in
+        storage/suggestion/gae_models.py.)
+
+        target_id: str. The ID of the target entity being suggested to.
+        target_version_at_submission: int. The version number of the target
+            entity at the time of creation of the suggestion.
+        author_id: str. The ID of the user who submitted the suggestion.
+        change_cmd: dict. The details of the suggestion.
+        score_category: str. The category to score the author on.
+        description: str. The description of the changes provided by the author.
+        assigned_reviewer_id: str(optional). The ID of the user assigned to
+            review the suggestion.
+        final_reviewer_id: str(optional). The ID of the reviewer who has
+            accepted/rejected the suggestion.
+    """
+
+    # TODO(nithesh): Remove the check for target type once the feedback threads
+    # are generalised for all types of entities. As at the moment feedback
+    # threads can only be linked to explorations, we have this check.
+    # This will be completed as a part of milestone 2 of the generalised review
+    # system project.
+    if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+        thread_id = feedback_services.create_thread(
+            target_id, None, author_id, description,
+            DEFAULT_SUGGESTION_THREAD_SUBJECT)
+        # This line will be removed after the feedback thread migration is
+        # complete and the IDs for both models match.
+        thread_id = suggestion_models.TARGET_TYPE_EXPLORATION + '.' + thread_id
+    if assigned_reviewer_id:
+        suggestion_models.SuggestionModel.create(
+            suggestion_type, target_type, target_id,
+            target_version_at_submission, suggestion_models.STATUS_IN_REVIEW,
+            author_id, assigned_reviewer_id, final_reviewer_id, change_cmd,
+            score_category, thread_id)
+    else:
```

Comment:
Find a way not to branch here. So much of this is the same between both branches. Define a variable for the status (`status = ... if ... else ...`) and use that.

In general, repeat yourself as little as possible. This is a very important principle in software engineering.

Reply:
Done. Thanks!